### PR TITLE
Updates Multi-Pool IPAM Docs for v1.14 Release

### LINKED ISSUE #26932: Test IPAM Multi-pool getting started guide with 1.14.0-rc.1
```[tasklist]
### Tasks
- [x] Install Cilium 1.14.0-rc.1
- [x] Test [IPAM Multi-pool getting started guide](https://docs.cilium.io/en/v1.14/network/kubernetes/ipam-multi-pool/#gsg-ipam-crd-multi-pool)
```

### PR DESCRIPTION
**Pull request by danehans**
- Updates the nginx image to the latest version (includes curl).
- Adds a step to test connectivity by curl'ing between pods.
- Adds a reference to https://github.com/cilium/cilium/issues/26966

Fixes: #26932

### CODE REVIEW

File: Documentation/network/concepts/ipam/multi-pool.rst [hunk 1]
```diff
@@ -70,7 +70,10 @@ PodCIDRs are allocated:
 New pools can be added at run-time. The list of CIDRs in each pool can also be
 extended at run-time. In-use CIDRs may not be removed from an existing pool, and
 existing pools may not be deleted if they are still in use by a Cilium node.
-The mask size of a pool is immutable and the same for all nodes.
+The mask size of a pool is immutable and the same for all nodes. Neither restriction
+is enforced until `Issue #26966`_ is resolved.
+
+.. _Issue #26966: https://github.com/cilium/cilium/issues/26966
```
Comment (joestringer) [inline Documentation/network/concepts/ipam/multi-pool.rst]: We have a sphinx extension for this:

```suggestion
is enforced until :gh-issue:`26966` is resolved.
```

File: Documentation/network/kubernetes/ipam-multi-pool.rst [hunk 2]
```diff
@@ -179,6 +186,8 @@ Validate installation
        [...]
        ✅ All 42 tests (295 actions) successful, 13 tests skipped, 0 scenarios skipped.
 
+  **Note:** The connectivity test requires a cluster with at least 2 worker nodes to complete successfully.
```
Comment (joestringer) [inline Documentation/network/kubernetes/ipam-multi-pool.rst:189]: Why is this required? I see that the example above tests from one node to another, but wouldn't it also work between local pods..?

**Review by joestringer — CHANGES_REQUESTED** [review]
Thanks for the PR. Minor nit on the multipool and a question later in the guide.

File: Documentation/network/kubernetes/ipam-multi-pool.rst (hunk 2 above)
Comment (danehans) [reply Documentation/network/kubernetes/ipam-multi-pool.rst:189]: The tests that run between pods on a node succeed. `cilium connectivity test` also runs tests that are expected to run on different nodes. The pods for these tests fail to get scheduled due to [taints and tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/).

File: Documentation/network/concepts/ipam/multi-pool.rst (hunk 1 above)
Comment (danehans) [reply Documentation/network/concepts/ipam/multi-pool.rst]: Thanks for the heads-up. Commit 9548223 resolves this comment.

**Review by joestringer — APPROVED** [review]

File: Documentation/network/kubernetes/ipam-multi-pool.rst (hunk 2 above)
Comment (tommyp1ckles) [reply Documentation/network/kubernetes/ipam-multi-pool.rst:189]: ~This is a general requirement of the connectivity tests, we should perhaps just be a warning when running the tests so we don't have to have similar comments everywhere.~

Edit: on recent versions of the cli this appears to already exist.

**Review by tommyp1ckles — APPROVED** [review]
looks good thanks!
